### PR TITLE
Add e2e test for connection timeout with l7-ilb

### PR DESCRIPTION
**Pull request by spencerhance**
blocked by #852 

/assign @freehan

### CODE REVIEW

File: cmd/e2e-test/timeout_test.go [hunk 1]
```diff
@@ -122,6 +122,100 @@ func TestTimeout(t *testing.T) {
...
+	}{
+		{
+			desc: "http with 60s timeout",
+			beConfig: fuzz.NewBackendConfigBuilder("", "backendconfig-1").
+				SetTimeout(42).
+				Build(),
+		},
+		{
+			desc: "http no timeout defined",
+			beConfig: fuzz.NewBackendConfigBuilder("", "backendconfig-1").
```
Comment (freehan) [inline cmd/e2e-test/timeout_test.go]: I am not sure if you still need this test case. 

I am wondering if you can just add a verifyTimeout on the Basic ILB test

Comment (freehan) [reply cmd/e2e-test/timeout_test.go]: Unless you want to test transitions. For instance,

Set Timeout -> Remove Timeout

Comment (spencerhance) [reply cmd/e2e-test/timeout_test.go]: Good call, I can remove it

**Pull request merged**